app/controllers/application_controller.rb

```diff
@@ -38,11 +36,13 @@ def redirect_unauthenticated
   end
 
   def set_current_user
-    @current_user =
+    Current.user =
       if session.key? :user_id
         User.find_by id: session[:user_id]
       elsif session.key? :spire
-        User.find_by spire: session[:spire]
+        User.find_by(spire: session[:spire]).tap do |user|
+          session[:user_id] = user&.id
```

Comment:
As far as I can tell, we've never set `session[:user_id]` before, so the `if` above was never used.